### DISCUSSION TITLE
1024-color sixel images

### DISCUSSION THREAD

**Discussion by 12Me21** [ImageMagick/ImageMagick#3018] [Development]
xterm (and likely other terminals) support up to 1024 colors in the sixel palette, while imagemagick only supports 256.  
I've modified sixel.c to increase the limit (required switching from `char` to `short`)  
[changes to sixel.c](https://github.com/ImageMagick/ImageMagick/compare/master...12Me21:master#diff-76512c4f7628167323415707d351f121dce4be938cf44a59dddf3662e7a0e3f7)  
I'm not sure 1024 colors should be the default, though (idk how widely supported it is, and the output is much larger of course)  
One issue that exists currently, but could become more of a problem with a larger palette: sixel uses 0-100 instead of 0-255 for each rgb channel (it also supports hsv which may be more precise?) so you might end up with duplicate colors in the palette.
Screenshot of 256 colors vs 1024 colors:  
![image](https://user-images.githubusercontent.com/4053256/102834436-087c1c80-43c2-11eb-8b1b-a82d8278a109.png)

**Comment by urban-warrior** [comment]
Thanks for the patch to the sixel format.  We'll merge it in the master branch within a day or two.
